tools/halide_benchmark.h

```diff
@@ -20,19 +59,69 @@ namespace Tools {
 // for real-world use. For now, callers using this to benchmark GPU
 // code should measure with extreme caution.
 
-template <typename F>
-double benchmark(int samples, int iterations, F op) {
-    double best = std::numeric_limits<double>::infinity();
-    for (int i = 0; i < samples; i++) {
-        auto t1 = std::chrono::high_resolution_clock::now();
-        for (int j = 0; j < iterations; j++) {
-            op();
+struct BenchmarkConfig {
+    // Terminate when the relative difference between the best runtime
+    // seen and the third-best runtime seen is no more than
+    // this. Controls accuracy. The closer to zero this gets the more
+    // reliable the answer, but the longer it may take to run.
+    double accuracy {0.03};
+    
+    // Take at least this much time per sample (seconds). Prevents
+    // benchmarking overhead and clock resolution being a major factor
+    // for very cheap benchmarks.
+    double min_time {0.005};
+
+    // Run at least this many iterations per sample.
+    uint64_t min_iters {1};
+
+    // Run at most this many iterations over all samples.
+    uint64_t max_iters {1000};
```

Comment:
This needs to default to a *much* larger value if total_iters really is supposed to track iterations over *all* samples